src/wasm/wasm-binary.cpp

```diff
@@ -5100,6 +5247,58 @@ void WasmBinaryReader::readDylink0(size_t payloadLen) {
   }
 }
 
+void WasmBinaryReader::readBranchHints(size_t payloadLen) {
+  auto sectionPos = pos;
+
+  auto numFuncs = getU32LEB();
+  for (Index i = 0; i < numFuncs; i++) {
+    auto funcIndex = getU32LEB();
+    auto& func = wasm.functions[funcIndex];
```

Comment:
We should guard against OOB indices here.

Reply:
Good point, done.